src/ucs/sys/sys.c

```diff
@@ -434,6 +434,21 @@ size_t ucs_get_page_size()
     return page_size;
 }
 
+int ucs_is_page_aligned(void *ptr)
+{
+    size_t page_size;
+
+    if (ptr) {
+        page_size = ucs_get_page_size();
+        if (ucs_likely(ucs_is_pow2(page_size))) {
```

Comment:
too complex - could assume page_size>0 (or assert it somewhere) , not need to "faster" handling of pow2. 
checking that page_size >0 implies it could be 0 to the innocent code reader